src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -86,16 +86,20 @@ private function doProcessValue($value, $isRoot = false)
             if ($ref = $this->getAutowiredReference($value)) {
                 return $ref;
             }
-            $message = $this->createTypeNotFoundMessage($value, 'it');
-
             if (ContainerBuilder::RUNTIME_EXCEPTION_ON_INVALID_REFERENCE === $value->getInvalidBehavior()) {
+                $container = new ContainerBuilder($this->container->getParameterBag());
+                $container->setAliases($this->container->getAliases());
+                $container->setDefinitions($this->container->getDefinitions());
+                $message = function () use ($container, $value) {
+                    return $this->createTypeNotFoundMessage($container, $value, 'it');
```

Comment:
`populateAvailableTypes` updates the state of the compiler pass. Doing that based on a different container might lead to weird behavior if we use that with multiple containers one after the other (well, maybe not worse than today if the container got mutated in the meantime though)

Reply:
yep I agree, but that's OK to me: since compiler passes has state, they cannot be shared between several containers. The shared state here is legit to me.